tools/serverpod_cli/lib/src/generator/dart/library_generators/entities_library_generator.dart

```diff
@@ -1369,6 +1402,330 @@ class SerializableEntityLibraryGenerator {
     }));
   }
 
+  Class? _buildEntityRepositoryClass(
+    String className,
+    List<SerializableEntityFieldDefinition> fields,
+    ClassDefinition classDefinition,
+  ) {
+    // TODO: remove when this class supports db operations
+    if (!(_hasAttachOperations(fields) || _hasDetachOperations(fields))) {
+      return null;
+    }
+
+    return Class((classBuilder) {
+      classBuilder
+        ..name = '${className}Repository'
+        ..constructors.add(Constructor((constructorBuilder) {
+          constructorBuilder
+            ..name = '_'
+            ..constant = true;
+        }))
+        ..fields.addAll([
+          if (_hasAttachOperations(fields))
+            Field((fieldBuilder) {
+              fieldBuilder
+                ..name = 'attach'
+                ..modifier = FieldModifier.final$
+                ..assignment = Code('const ${className}AttachRepository._()');
+            }),
+          if (_hasDetachOperations(fields))
+            Field((fieldBuilder) {
+              fieldBuilder
+                ..name = 'detach'
+                ..modifier = FieldModifier.final$
+                ..assignment = Code('const ${className}DetachRepository._()');
+            }),
+        ]);
+    });
+  }
+
+  Class? _buildEntityAttachRepositoryClass(
+    String className,
+    List<SerializableEntityFieldDefinition> fields,
+    ClassDefinition classDefinition,
+  ) {
+    if (!_hasAttachOperations(fields)) return null;
```

Comment:
Suggestion: Move check to call site.

Same reasoning here, since this could result in a null opt I think it belongs to logical decision belongs to the abstraction layer above (or the method would need to be named maybeBuildEntityAttachRepositoryClass).